rust_src/src/eval_macros.rs

```diff
@@ -129,7 +129,100 @@ macro_rules! def_lisp_sym {
+/// globals.h; the plain name should be used in the code.
+///
+/// E.g., the global "cons_cells_consed" is declared as "int
+/// f_cons_cells_consed" in globals.h, but there is a define:
+///
+///    #define cons_cells_consed globals.f_cons_cells_consed
+///
+/// All C code uses the `cons_cells_consed' name.  This is all done
+/// this way to support indirection for multi-threaded Emacs.
+#[macro_export]
+macro_rules! defvar_lisp {
+    ($fvname:ident, $lname:expr, $value:expr) => {
+        {
+            static mut o_fwd: ::remacs_sys::Lisp_Objfwd = ::remacs_sys::Lisp_Objfwd {
+                ty: ::remacs_sys::Lisp_Fwd_Obj,
+                objvar: ::std::ptr::null_mut()
+            };
+            unsafe { ::remacs_sys::defvar_lisp(&mut o_fwd,
+                                               $lname.as_ptr() as *const i8,
+                                               &mut ::remacs_sys::globals.$fvname);
+            }
+            unsafe { ::remacs_sys::globals.$fvname = $value; }
+        }
+    };
+}
+#[macro_export]
+macro_rules! defvar_lisp_nopro {
+    ($fvname:ident, $lname:expr, $value:expr) => {
+        {
+            static mut o_fwd: ::remacs_sys::Lisp_Objfwd = ::remacs_sys::Lisp_Objfwd {
+                ty: ::remacs_sys::Lisp_Fwd_Obj,
+                objvar: &mut ::remacs_sys::globals.$fvname,
+            };
+            unsafe { ::remacs_sys::defvar_lisp_nopro(&mut o_fwd,
+                                                     $lname.as_ptr() as *const i8,
+                                                     &mut ::remacs_sys::globals.$fvname); }
+            unsafe { ::remacs_sys::globals.$fvname = $value; }
+        }
+    };
+}
+#[macro_export]
+macro_rules! defvar_bool {
+    ($fvname:ident, $lname:expr, $value:expr) => {
+        {
+            static mut o_fwd: ::remacs_sys::Lisp_Boolfwd = ::remacs_sys::Lisp_Boolfwd {
+                ty: ::remacs_sys::Lisp_Fwd_Bool,
+                boolvar: &mut ::remacs_sys::globals.$fvnaem,
+            };
+            unsafe { ::remacs_sys::defvar_bool(&mut o_fwd,
+                                               $lname.as_ptr() as *const i8,
+                                               &mut ::remacs_sys::globals.$fvname); }
+            unsafe { ::remacs_sys::globals.$fvname = $value; }
+        }
+    };
+}
+#[macro_export]
+macro_rules! defvar_int {
+    ($fvname:ident, $lname:expr, $value:expr) => {
+        {
+            static mut o_fwd: ::remacs_sys::Lisp_Intfwd = ::remacs_sys::Lisp_Intfwd {
+                ty: ::remacs_sys::Lisp_Fwd_Int,
+                intvar: &mut ::remacs_sys::globals.$fvnaem,
+            };
+            unsafe { ::remacs_sys::defvar_int(&mut o_fwd,
+                                              $lname.as_ptr() as *const i8,
+                                              &mut ::remacs_sys::globals.$fvname); }
+            unsafe { ::remacs_sys::globals.$fvname = $value; }
+        }
+    };
+}
+// think about this later, since it uses offsetof
```

Comment:
I'm not a fan of commented-out code, it tends to bit-rot. Could we move this to an issue instead?

Reply:
sure